vscode/quint-vscode/server/src/definitions.ts

```diff
@@ -0,0 +1,36 @@
+import { Loc, ParserPhase3 } from "@informalsystems/quint"
+import { findName } from "./reporting"
+import { Position } from "vscode-languageserver/node"
+import { URI } from "vscode-uri"
+
+interface QuintDefinitionLink {
+  nameId: bigint,
+  name: string,
+  definitionId?: bigint,
+}
+
+export function findDefinition(
+  parsedData: ParserPhase3, uri: string, position: Position
+): QuintDefinitionLink | undefined {
+  const { modules, sourceMap, table } = parsedData
+  const results: [Loc, bigint][] = [...sourceMap.entries()].map(([id, loc]) => [loc, id])
```

Comment:
Would the following piece copy the whole array (presumably, a large one), even though we need just a few entries?

```ts
[...sourceMap.entries()]
```

Reply:
Yeah, I think I tried to generalize things to early here, we shouldn't need to convert those maps into lists of tuples at all. Changing this will require some meaningful refactor tho, as this `results` format is being used across different functions. I'll add an issue for this refactor.